File: smoke_tests/typescript/weak_cipher.ts
// Weak Cipher (DES/RC4/ECB) Test Cases

import * as crypto from 'crypto';

// Test 1: DES encryption
function encryptWithDES(data: string, key: Buffer): string {
    // VULNERABLE: DES is obsolete and insecure
    const cipher = crypto.createCipheriv('des', key, Buffer.alloc(8));
    return cipher.update(data, 'utf8', 'hex') + cipher.final('hex');
}

// Test 2: 3DES encryption
function encryptWith3DES(data: string, key: Buffer): string {
    // VULNERABLE: 3DES is deprecated and slow
    const cipher = crypto.createCipheriv('des-ede3', key, Buffer.alloc(8));
    return cipher.update(data, 'utf8', 'hex') + cipher.final('hex');
}

// Test 3: RC4 encryption
function encryptWithRC4(data: string, key: Buffer): string {
    // VULNERABLE: RC4 has known vulnerabilities
    const cipher = crypto.createCipheriv('rc4', key, null);
    return cipher.update(data, 'utf8', 'hex') + cipher.final('hex');
}

// Test 4: AES-ECB mode
function encryptWithECB(data: string, key: Buffer): string {
    // VULNERABLE: ECB mode is not semantically secure
    const cipher = crypto.createCipheriv('aes-128-ecb', key, null);
    return cipher.update(data, 'utf8', 'hex') + cipher.final('hex');
}

// Test 5: AES-256-ECB mode
function encryptWithAES256ECB(data: string, key: Buffer): string {
    // VULNERABLE: ECB mode leaks patterns even with AES-256
    const cipher = crypto.createCipheriv('aes-256-ecb', key, null);
    return cipher.update(data, 'utf8', 'hex') + cipher.final('hex');
}

File: smoke_tests/rust/weak_cipher.rs
// Weak Cipher (DES/RC4/ECB) Test Cases

// Test 1: DES encryption
fn encrypt_with_des(data: &[u8], key: &[u8]) -> Vec<u8> {
    // VULNERABLE: DES is obsolete and insecure
    // In a real scenario, this would use a DES implementation
    // Placeholder to demonstrate the vulnerability pattern
    let mut encrypted = data.to_vec();
    // Simulated DES encryption
    encrypted
}

// Test 2: RC4 encryption
fn encrypt_with_rc4(data: &[u8], key: &[u8]) -> Vec<u8> {
    // VULNERABLE: RC4 has known vulnerabilities
    // Placeholder for RC4 implementation
    let mut encrypted = data.to_vec();
    encrypted
}

// Test 3: AES-ECB mode encryption
fn encrypt_with_ecb(plaintext: &[u8], key: &[u8; 16]) -> Vec<u8> {
    use aes::Aes128;
    use block_modes::{BlockMode, Ecb};
    use block_modes::block_padding::Pkcs7;

    type Aes128Ecb = Ecb<Aes128, Pkcs7>;

    // VULNERABLE: ECB mode is not semantically secure
    let cipher = Aes128Ecb::new_from_slices(key, &[]).unwrap();
    cipher.encrypt_vec(plaintext)
}

// Test 4: 3DES usage
fn encrypt_with_3des(data: &[u8], key: &[u8]) -> Vec<u8> {
    // VULNERABLE: 3DES is deprecated and slow
    // Placeholder for 3DES implementation
    let mut encrypted = data.to_vec();
    encrypted
}

// Test 5: Blowfish with small key
fn encrypt_with_weak_blowfish(data: &[u8], key: &[u8; 8]) -> Vec<u8> {
    // VULNERABLE: Small key size for Blowfish
    // Placeholder for Blowfish implementation
    let mut encrypted = data.to_vec();
    encrypted
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_weak_ciphers() {
        let data = b"sensitive data";
        let key = b"secretkey";
        let _ = encrypt_with_des(data, key);
        let _ = encrypt_with_rc4(data, key);
    }
}
